src/Streamly/Prelude.hs

```diff
@@ -138,6 +142,31 @@ iterateM step = fromStream . go
        a <- step s
        yld s (Just (go a))
 
+-- | @repeat a@ creates an infinite stream of @a@.
+repeat :: (Streaming t) => a -> t m a
+repeat a = a .: repeat a
+
+-- | @replicate n a@ creates an @n@ element stream of only @a@.
+replicate :: (Streaming t) => Int -> a -> t m a
+replicate n a = take n $ repeat a
+
+-- | @cycle m@ creates an infinite stream by starting at the first element of
+-- @m@ after exhausting all other elements.
+-- The given stream may NOT be empty.
+cycle :: (Streaming t) => t m a -> t m a
+cycle m = fromStream $ goIfNotNull $ toStream m
+    where
+    goIfNotNull m1 = Stream $ \svr _ yld ->
+        let stop = error "Can't cycle an empty Stream"
```

Comment:
Partial functions are to be avoided. But I did not comment  on this because I am not in favor of having the `cycle` function in the first place as there is a `cycle1` in `Data.Semigroup` that we can use for the same purpose.